opentera-webrtc-native-client/OpenteraWebrtcNativeClient/include/OpenteraWebrtcNativeClient/Utils/FunctionTask.h

```diff
@@ -10,23 +9,10 @@
 namespace opentera
 {
     template<class T>
-    class FunctionTask : public webrtc::QueuedTask
+    class FunctionTask
     {
-        std::function<T()> m_function;
-        rtc::Event m_event;
-        T m_returnedValue;
-
     public:
-        FunctionTask(const std::function<T()>& function) : m_function(function) {}
-
-        ~FunctionTask() override = default;
-
-        bool Run() override
-        {
-            m_returnedValue = std::move(m_function());
-            m_event.Set();
-            return false;
-        }
+        FunctionTask() = delete;
```

Comment:
Class is not used.
While breaking the API, why not move that to a free function, like this:
```cpp
namespace function_task
{
    template <typename T>
    T callSync(...) {...}

    template <>
    void callSync<void>(...) {...}
}
```